antivirus.py

```diff
@@ -1,17 +1,188 @@
+REVISED_KW_SCORE_MAP = {}
+
+
+class AntiVirusHost:
+    def __init__(self, name: str, ip: str, port: int, method: str, endpoint: str) -> None:
+        if method not in VALID_METHODS:
+            raise ValueError(f"Given method '{method}' is not one of {VALID_METHODS}.")
+
+        self.name = name
+        self.ip = ip
+        self.port = port
+        self.method = method
+        self.endpoint = endpoint
+        self.client = IcapClient(
+            host=self.ip,
+            port=self.port,
+            respmod_service=self.endpoint
+        ) \
+            if self.method == ICAP_METHOD else Session()
+        self.sleeping = False
+
+    def __eq__(self, other):
+        return self.name == other.name and self.ip == other.ip and \
+               self.port == other.port and self.method == other.method and \
+               self.endpoint == other.endpoint and type(self.client) == type(other.client) and \
+               self.sleeping == other.sleeping
+
+    def sleep(self, timeout: int) -> None:
+        self.sleeping = True
+        sleep(timeout)
+        self.sleeping = False
+
+
+class AvHitSection(ResultSection):
+    def __init__(self, av_name: str, virus_name: str, engine: Dict[str, str], heur_id: int) -> None:
+        title = f"{av_name} identified the file as {virus_name}"
+        json_body = dict(
+            av_name=av_name,
+            virus_name=virus_name,
+            scan_result="infected" if heur_id == 1 else "suspicious",
+            engine_version=engine['version'] if engine else "unknown",
+            engine_definition_time=engine['def_time'] if engine else "unknown",
+        )
+
+        super(AvHitSection, self).__init__(
+            title_text=title,
+            body_format=BODY_FORMAT.KEY_VALUE,
+            body=json.dumps(json_body),
+        )
+        signature_name = f'{av_name.upper()}.{virus_name}'
+        section_heur = Heuristic(heur_id)
+        if signature_name in REVISED_SIG_SCORE_MAP:
+            section_heur.add_signature_id(signature_name, REVISED_SIG_SCORE_MAP[signature_name])
+        elif any(kw in signature_name.lower() for kw in REVISED_KW_SCORE_MAP):
+            section_heur.add_signature_id(
+                signature_name,
+                max([REVISED_KW_SCORE_MAP[kw] for kw in REVISED_KW_SCORE_MAP if kw in signature_name.lower()])
+            )
+        else:
+            section_heur.add_signature_id(signature_name)
+        self.heuristic = section_heur
+        self.add_tag('av.virus_name', virus_name)
+        if heur_id == 2:
+            self.add_tag("av.heuristic", virus_name)
+
+        # TODO: Isolate parts of X-Virus-ID/virus_name according https://encyclopedia.kaspersky.com/knowledge/rules-for-naming/
+        # So that we can tag more items of interest
 
 
 class AntiVirus(ServiceBase):
     def __init__(self, config: Optional[Dict] = None) -> None:
         super(AntiVirus, self).__init__(config)
+        self.hosts: Optional[List[AntiVirusHost]] = None

```

Comment:
Using an empty list as the default and dropping Optional from the type hint would help static type checkers in inferring types since self.hosts should never be None when executing.